method.rb

```diff
@@ -0,0 +1,248 @@
+      puts 'warning: given block not used'
+      loop do
+        status_cumulator &&= (self[i] == parameter)
+        i += 1
+        break if i == size || status_cumulator == false
+      end
+    elsif bg == false && parameter.nil?
+      loop do
+        break if i == size || status_cumulator == false
+
+        status_cumulator &&= !self[i].nil?
+        i += 1
+      end
+    elsif bg == false && parameter.is_a?(Regexp)
+      loop do
+        status_cumulator &&= !parameter.match(self[i]).nil?
+        i += 1
+        break if i == size || status_cumulator == false
+      end
+    elsif bg == false && parameter.is_a?(Class)
+      loop do
+        status_cumulator &&= self[i].is_a?(parameter)
+        i += 1
+        break if i == size || status_cumulator == false
+      end
+    else
+      loop do
+        status_cumulator &&= (self[i] == parameter)
+        i += 1
+        break if i == size || status_cumulator == false
+      end
+    end
+    status_cumulator
+  end
+
+  def my_any?(parameter = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/MethodLength, Metrics/AbcSize
+    status_cumulator = false
+    i = 0
+    bg = block_given?
+    if bg == true && parameter.nil?
+      loop do
+        status_cumulator ||= yield self[i]
+        i += 1
+        break if i == size || status_cumulator == true
+      end
+    elsif bg == true && parameter.is_a?(Regexp)
+      puts 'warning: given block not used'
+      loop do
+        status_cumulator ||= !parameter.match(self[i]).nil?
+        i += 1
+        break if i == size || status_cumulator == true
+      end
+    elsif bg == true && parameter.is_a?(Class)
+      puts 'warning: given block not used'
+      loop do
+        status_cumulator ||= self[i].is_a?(parameter)
+        i += 1
+        break if i == size || status_cumulator == true
+      end
+    elsif bg == true && parameter_condition?(parameter)
+      puts 'warning: given block not used'
+      loop do
+        status_cumulator ||= (self[i] == parameter)
+        i += 1
+        break if i == size || status_cumulator == true
+      end
+    elsif bg == false && parameter.nil?
+      loop do
+        break if i == size || status_cumulator == true
+
+        status_cumulator ||= !self[i].nil?
+        i += 1
+      end
+    elsif bg == false && parameter.is_a?(Regexp)
+      loop do
+        status_cumulator ||= !parameter.match(self[i]).nil?
+        i += 1
+        break if i == size || status_cumulator == true
+      end
+    elsif bg == false && parameter.is_a?(Class)
+      loop do
+        status_cumulator ||= self[i].is_a?(parameter)
+        i += 1
+        break if i == size || status_cumulator == true
+      end
+    else
+      loop do
+        status_cumulator ||= (self[i] == parameter)
+        i += 1
+        break if i == size || status_cumulator == true
+      end
+    end
+    status_cumulator
+  end
+
+  def my_none?(parameter = nil, &my_block)
```

Comment:
**my_none** when no block or argument is given it should return true only if none of the collection members is true